annotation-processor/src/main/java/org/hibernate/validator/ap/classchecks/ClassCheck.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.ap.classchecks;
+
+import java.util.Collection;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+
+import org.hibernate.validator.ap.checks.ConstraintCheckError;
+
+/**
+ * <p>
+ * Implementations represent checks, which determine whether a given class
+ * is implemented in correct way.
+ * </p>
+ * <p>
+ * Implementations should be derived from {@link AbstractClassCheck} in
+ * order to implement only those check methods applicable for the element kinds
+ * supported by the check.
+ * </p>
+ *
+ *
+ * @author Marko Bekhta
+ */
+public interface ClassCheck {
+
+	/**
+	 * Checks, whether the given method is written correctly.
+	 *
+	 * @param element A method under investigation.
+	 *
+	 * @return A collection with errors, that describe, why the given method
+	 *         is not correctly implemented. In case no errors occur (the
+	 *         method is written correctly), an empty set must be returned.
+	 */
+	Collection<ConstraintCheckError> checkMethod(ExecutableElement element);
+
+
+	/**
+	 * run all checks on the element
+	 *
+	 * @param element an element under investigation;
```

Comment:
No ; at the end (and according to http://hibernate.org/validator/contribute/#javadoc, no dot either for @param and @return - note that the existing code is not very consistent about that).